bhklab/CoreGx
# development

- Updated coreGx.Rmd vignette to reflect recent changes in class strucutre
- Renamed vignette to TreatmentResponseExperiment to indicate new package nomenclature and added updates class and structural diagrams
- Delete old LongTable vignette file and started updating NEWS.md
- Add documentation object for TREDataMapper-accessors
- Fix incorrect method signature in rowData and colData methods for TREDataMapper class
- Replaced straggler use of LongTable instead of TreatmentResponseExperiment in TRE vignette
- Removed v 2.1.2 from NEWS before merging with master
- Fix missed merge conflict in DESCRIPTION and incremented version number
- Added metaConstruct method for TREDataMapper object; updated TreatmentResponseExperiment constructor to match LongTable function arguments
- Tests added for LongTable accessor&utils
- Minor fix: subset i rowname length check works now
- Fix #146: join on wrong index in reindex,LongTable-method was failing to update the assayIndex row and colKeys
- Renamed all instances of lt to tre in tests to represent updated package nomenclature
- Add warning for duplciated idColumns in rowData assignment method; stil need to propogate change to colData
- New test cases for subset method and TRE class added
- Added NCI-ALMANAC TRE subset to the package.
- Fix duplicated detection in rowData method
- New subset() tests added. Compressed test dataset. Added missing colData param check for LongTable consturctor.
- Split unit tests by .R files.
- Basic aggregation of TreatmentResponseExperiment now working
- Refactored row and colData<-,LongTable-method to use a shared helper method
- Fix #148 by feeding back rowKeya nd/colKey from assays to index after subset
- fix #148 by feeding back rowKeya nd/colKey from assays to index after subset
- Debugging LongTable/TRE unit tests
- Add .table_is_subset utility for use in unit test
- Fix unit tests for subset,LongTable-method to account for the fact that additional row or colKeys may get dropped when there are no observations remaining in the table
- Update unit tests to account for change in how subsetting a LongTable works
- All unit tests now passing with modified subset,LongTable-method algorithm
- Forgot to git add some stuf f in last push
- Fix bug in aggregate2 causing default name parsing to fail when no aggregation calls have names
- Add tests to ensure aggregate2 results are identical to data.table aggregation
- Add test to ensure aggregate,LongTable-method is equivalent to data.table aggregation of the associated assay
- Add test to ensure that automatic column name generation works in aggregate2 and aggreagate,LongTable-methods
- Add tests to ensure assignment of summarized assays works correctly; fix #147; add test case for #147
- Debugging issue with assignment of summary assays
- Update .subsetByIndex to apply unique to each assay data.table before assigning back to the TRE object; this fixes #149
- Updated comments for .subsetByIndex
- All unit tests passing
